web/src/app/projects/view/view.component.ts

```diff
@@ -28,57 +34,73 @@ export class ViewProjectComponent implements OnInit {
         private dialog: MatDialog,
         private projectService: ProjectService,
         private repositoryService: RepositoryService,
-        private authService: AuthenticationService,
+        private authService: AuthenticationService
     ) {
         this.project.uid = this.route.snapshot.paramMap.get('uid');
     }
 
     ngOnInit(): void {
         this.projectSubscription = this.projectService
             .findOneById(this.project.uid)
-            .subscribe((project: ProjectModel) => this.project = project);
+            .subscribe((project: ProjectModel) => {
+                this.project = project;
+                // Added check for not authenticated user and private project details
+                if (project.type === 'private' && !this.authService.isAuthenticated) {
```

Comment:
Need to check if they are the owner also?

Reply:
For viewing the public projects we need not to check if they are owner also. Any one can see the project right but in case of edit project only owner is allowed to edit the public and private projects.

Reply:
👍